core-web/libs/template-builder/src/lib/directives/container-options.directive.ts

```diff
@@ -0,0 +1,80 @@
+import { Observable, Subscription } from 'rxjs';
+
+import { Directive, OnDestroy, OnInit, Optional, Self } from '@angular/core';
+
+import { Dropdown } from 'primeng/dropdown';
+
+import { debounceTime, map, switchMap, tap } from 'rxjs/operators';
+
+import { DotContainersService } from '@dotcms/data-access';
+import { DotContainerEntity, DotDropdownSelectOption } from '@dotcms/dotcms-models';
+
+const DEFAULT_LABEL_NAME_INDEX = 'label';
+const DEFAULT_VALUE_NAME_INDEX = 'value';
+
+/**
+ * Directive to set a default configuration of Dropdown or MultiSelect (PrimeNG) and translate the label of the options
+ *
+ * @export
+ * @class DotDropdownDirective
+ */
+@Directive({
+    selector: 'p-dropdown[dotcmsContainerOptions]',
+    standalone: true
+})
+export class ContainerOptionsDirective implements OnInit, OnDestroy {
+    private readonly control: Dropdown;
+    private readonly maxOptions = 10;
+    private onFilterSubscription: Subscription;
+
+    constructor(
+        @Optional() @Self() private readonly primeDropdown: Dropdown,
+        private readonly dotContainersService: DotContainersService
+    ) {
+        this.control = this.primeDropdown;
+
+        if (this.control) {
+            this.control.optionLabel = DEFAULT_LABEL_NAME_INDEX;
+            this.control.optionValue = DEFAULT_VALUE_NAME_INDEX;
+        } else {
+            console.warn('ContainerOptionsDirective is for use with PrimeNg Dropdown');
+        }
+    }
+
+    ngOnInit() {
+        this.fetchContainerOptions('').subscribe((options) => {
```

Comment:
```suggestion
        this.fetchContainerOptions().subscribe((options) => {
```

You can do the default in the method.

Reply:
Does this also needs a `take(1)`?